src/resources/worker.py

```diff
@@ -120,7 +122,17 @@ async def run_worker(charm_config, supported_workflows, supported_activities, mo
         client_config.auth = AuthOptions(provider=charm_config["auth-provider"], config=_get_auth_header(charm_config))
 
     if charm_config["encryption-key"].strip() != "":
-        client_config.encryption = EncryptionOptions(key=charm_config["encryption-key"])
+        client_config.encryption = EncryptionOptions(key=charm_config["encryption-key"], compress=True)
+
+    interceptors = []
```

Comment:
wondering if this should be part of the lib, and not this

Reply:
The interceptor for the client is different to the intereceptor for the worker, activity interceptors are implemented at the worker-level.

Reply:
After our conversation, I've updated our Temporal lib to also export a Worker which has a configurable Sentry interceptor. The charm now uses this worker.